csrc/device_lower/analysis/circular_buffer.cpp

```diff
@@ -346,6 +346,10 @@ bool hasIndependentWarpGroups(const TensorView* tv) {
 
   // Step 5: Use independent warp groups if warp specialized axis is to the
   // left of the stage_slice_position
+  NVF_ERROR(
```

Comment:
We have to have the `stage_slice_position` to the right of the warp-specialized axis for ping-pong matmuls.

Reply:
We parallelize output tiles with warp-specialized axis but tile along the k axis of the input operands.

Reply:
The err msg is confusing, should be `stage_slice_position either stays to the left of or immediately follows the warp-specialized domain.`, so we still allow `stage_slice_position` to the right of the warp-specialized axis.

Reply:
> We have to have the `stage_slice_position` to the right of the warp-specialized axis for ping-pong matmuls.

Do you mean must use `stage_slice_position_4` for matmul? If that's the case, I can revise this limitation to apply only to 1D TMA with predicate. But the mbarrier issue seems universial for both normalization and matmul.

Reply:
`matmul` inserts mbarriers to enforce an ordering between tensor-cores and cuda cores. I suppose we can prohibit this if matmuls are not detected in the kernel.